cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -0,0 +1,426 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/detail/utilities/assert.cuh>
+#include <cudf/detail/utilities/hash_functions.cuh>
+#include <cudf/sorting.hpp>
+#include <cudf/table/row_operators.cuh>
+#include <cudf/table/table_device_view.cuh>
+#include <cudf/utilities/traits.hpp>
+#include <cudf/utilities/type_dispatcher.hpp>
+
+#include <thrust/equal.h>
+#include <thrust/swap.h>
+#include <thrust/transform_reduce.h>
+
+#include <cuda/std/tuple>
+#include <cuda/std/utility>
+
+#include <limits>
+#include <memory>
+#include <utility>
+
+namespace cudf {
+namespace experimental {
+
+/**
+ * @brief A map from cudf::type_id to cudf type that excludes LIST and STRUCT types.
+ *
+ * To be used with type_dispatcher in place of the default map, when it is required that STRUCT and
+ * LIST map to void.
+ *
+ * Usage:
+ * @code
+ * type_dispatcher<non_nested_id_to_type>(data_type(), functor{});
+ * @endcode
+ */
+template <cudf::type_id t>
+struct non_nested_id_to_type {
+  using type = std::conditional_t<cudf::is_nested(data_type(t)), void, id_to_type<t>>;
+};
+
+namespace lex {
+
+/**
+ * @brief Performs a relational comparison between two elements in two columns.
+ *
+ * @tparam Nullate A cudf::nullate type describing how to check for nulls.
+ */
+template <typename Nullate>
+class element_comparator {
+ public:
+  /**
+   * @brief Construct type-dispatched function object for performing a
+   * relational comparison between two elements.
+   *
+   * @note `lhs` and `rhs` may be the same.
+   *
+   * @param has_nulls Indicates if either input column contains nulls.
+   * @param lhs The column containing the first element
+   * @param rhs The column containing the second element (may be the same as lhs)
+   * @param null_precedence Indicates how null values are ordered with other values
+   * @param depth The depth of the column if part of a nested column @see preprocessed_table::depths
+   */
+  __host__ __device__ element_comparator(Nullate has_nulls,
+                                         column_device_view lhs,
+                                         column_device_view rhs,
+                                         null_order null_precedence,
+                                         int depth = 0)
+    : _lhs{lhs}, _rhs{rhs}, _nulls{has_nulls}, _null_precedence{null_precedence}, _depth{depth}
+  {
+  }
+
+  __host__ __device__ element_comparator(Nullate has_nulls,
+                                         column_device_view lhs,
+                                         column_device_view rhs)
+    : _lhs{lhs}, _rhs{rhs}, _nulls{has_nulls}
```

Comment:
I don't know why this ctor (and [this](https://github.com/rapidsai/cudf/blob/cbc4d8bf8670819cb5415224db0a428c663db08d/cpp/include/cudf/table/row_operators.cuh#L289-L294), from where it's copied) exists, instead of just using default value for `null_precedence`. @davidwendt would know because it's introduced in nullate PR

Reply:
I don't see where it is used. Looks like it can be removed.